tests/signature/test_signature_spec.lua

```diff
@@ -50,46 +51,52 @@ describe(" Signature Methods Tests ", function()
 		url.git@github.com:.pushinsteadof=http://github.com/
 		user.name=TEST_USER
 		user.email=testusernoreply@github.com
-		
+ 
 	]]--
-
-	it("Tests signature_default method",function()
-		-- Get the Default signature for repository
-		local sign_default = lib.luagit2_signature_default(repo)
-		assert.are.equal("TEST_USER", lib.luagit2_get_signature_details(sign_default,1))
-		assert.are.equal("testusernoreply@github.com", lib.luagit2_get_signature_details(sign_default,2))
-		lib.luagit2_signature_free(sign_default)
-	end)
-
-	it("Tests signature from string method",function()
-		-- Create a sign from a complete signature string
-		-- as you see in a commit.
-		local sign_string = "Testing_user <something@example.com> 1461698487 +0200"
-		local sign_from_string_buffer = lib.luagit2_signature_from_buffer(sign_string)
-
-		assert.are.equal("Testing_user", lib.luagit2_get_signature_details(sign_from_string_buffer,1))
-		assert.are.equal("something@example.com", lib.luagit2_get_signature_details(sign_from_string_buffer,2))
-		-- free the signature
-		lib.luagit2_signature_free(sign_from_string_buffer)
-	end)
-
-	it("Tests signature now method",function()
-		local sign_now = lib.luagit2_signature_now("new_user","new_user@example.com")
-
-		assert.are.equal("new_user", lib.luagit2_get_signature_details(sign_now,1))
-		assert.are.equal("new_user@example.com", lib.luagit2_get_signature_details(sign_now,2))
-		-- free the signature
-		lib.luagit2_signature_free(sign_now)
-	end)
-
-	it("Tests signature dup method",function()
-		local sign_now = lib.luagit2_signature_now("new_user","new_user@example.com")
-		local sign_dup = lib.luagit2_signature_dup(sign_now)
-
-		assert.are.equal("new_user", lib.luagit2_get_signature_details(sign_dup,1))
-		assert.are.equal("new_user@example.com", lib.luagit2_get_signature_details(sign_dup,2))
-		-- free the signatures
-		lib.luagit2_signature_free(sign_now)
-		lib.luagit2_signature_free(sign_dup)
-	end)
+    
+    it("Tests signature_default method", function()
+        -- Get the Default signature for repository
+        local sign_default = luagit2.signature_default(repo)
+        local name, email = luagit2.get_signature_details(sign_default)
+        
+        assert.are.equal("TEST_USER", name)
```

Comment:
this is much more organised! well done 👍

Reply:
thankyou :smile: